### PR TITLE
Kamoser/igp int test

### PR DESCRIPTION
**Pull request by KyleMoser**

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -630,11 +630,12 @@ func NewSimApp(
 	)
 
 	// TODO: How are the domains registered selected.. using 12345 as a placeholder
```
Comment (DavidNix) [inline app/app.go:632]: This is a good question. Wondering that myself.

File: x/mailbox/keeper/keeper.go [hunk 2]
```diff
@@ -45,20 +48,33 @@ func NewKeeper(cdc codec.BinaryCodec, key storetypes.StoreKey, cwKeeper *cosmwas
...
 		version:     0,
-		domain:      domain,
 		pcwKeeper:   cosmwasm.NewDefaultPermissionKeeper(cwKeeper),
 		Tree:        &imt.Tree{},
 		Delivered:   map[string]bool{},
 	}
 }
 
-func (k Keeper) VerifyMessage(messageBytes []byte) (string, error) {
+func (k *Keeper) SetDomain(c context.Context, domain uint32) {
```
Comment (misko9) [inline x/mailbox/keeper/keeper.go:57]: It is still unclear to me why the domain needs to be in the KVStore. It adds r/w operations.

File: x/mailbox/keeper/genesis.go [hunk 3]
```diff
@@ -26,6 +26,8 @@ func (k Keeper) InitGenesis(ctx sdk.Context, gs types.GenesisState) error {
 	for _, msgDelivered := range gs.DeliveredMessages {
 		k.Delivered[msgDelivered.Id] = true
 	}
+
+	k.SetDomain(ctx, gs.Domain)
```
Comment (misko9) [inline x/mailbox/keeper/genesis.go:30]: How will the domain initially be set for existing chains?

**Review by misko9 — COMMENTED** [review]
looking good!

File: x/mailbox/keeper/keeper.go (hunk 2 above)
Comment (misko9) [reply x/mailbox/keeper/keeper.go:57]: Without it in the genesis file, why doesn't it persist in memory when other items like the authority do?

File: x/mailbox/keeper/genesis.go (hunk 3 above)
Comment (misko9) [reply x/mailbox/keeper/genesis.go:30]: App initialization doesn't have a context. So would it need to be set in an upgrade handler?

**Pull request merged**